unified_ecommerce/authentication.py

```diff
@@ -76,3 +81,59 @@ def authenticate(self, request):
             return (user, None)
 
         return None
+
+
+class ApiGatewayAuthentication(BaseAuthentication):
+    """
+    Handles authentication when behind an API gateway.
+
+    If the app is sitting in front of something like APISIX, the app will get
+    authentication information in the HTTP_X_USERINFO header. So, this decodes that,
+    and then attempts to authenticate the user or creates them if they don't exist.
+    """
+
+    def authenticate(self, request):
+        """Authenticate the user based on HTTP_X_USERINFO."""
+
+        user_info = request.META.get("HTTP_X_USERINFO", False)
+
+        if not user_info:
+            return None
+
+        decoded_user_info = json.loads(base64.b64decode(user_info))
+
+        logger.info(
+            "ApiGatewayAuthentication: Checking for existing user for %s",
+            decoded_user_info["preferred_username"],
```

Comment:
Nit: username and email are pulled out several times in this function, it's probably best to turn these into local variables instead.